api/src/services/user-service.ts

```diff
@@ -14,7 +14,19 @@ export class UserService {
 
 	searchUsers = async (userSearch: UserSearch): Promise<User[]> => {
 		const usersPersistence = await userDb.searchUsers(userSearch);
-		return usersPersistence.map((user) => this.getUserObject(user));
+		return usersPersistence
+			.map((user) => {
+				const userObj = this.getUserObject(user);
+				if (!userObj) {
+					return null;
+				}
+				return {
+					...userObj,
+					verifiableCredentials: undefined,
+					claim: userObj.isPrivate ? undefined : userObj.claim
```

Comment:
instead of setting `claim` to `undefined` you can just remove it from the `userObj`
```
userObj?.isPrivate && delete userObj?.claim;
```

Reply:
I don't know how this should work in the return object creation, I would need another line having an if check.

Reply:
yes, it needs to be done prior to the return statement

Reply:
Ok I do prefer undefined in this case